R/Artifact.R

```diff
@@ -22,11 +22,30 @@ ArtifactRecord <- R6::R6Class( # nolint object_name_linter
       load_file(file_path, suffix, ...)
     },
     #' @description
-    #' Cache the artifact to the local filesystem. This currently only supports
-    #' S3 storage.
+    #' Cache the artifact to the local filesystem. When the Python `lamindb`
+    #' package is not available this only supports S3 storage.
     #'
     #' @return The path to the cached artifact
     cache = function() {
+
+      py_lamin <- private$.instance$get_py_lamin()
+      if (!is.null(py_lamin)) {
```

Comment:
Is there every a case in which the Python package is not available? We already needed it for `lamin connect` etc. -- so I think it's always going to be available?

Reply:
Not necessarily -- if the user provides a correct `current_user.env` they could still connect to a lamindb instance without the python client. they just won't be able to track, create or delete records at this stage.

Reply:
I guess you need to CLI installed somewhere to do that but it doesn't necessarily mean it's an environment that **{reticulate}** can find.

Reply:
Ok!